src/linear_algebra/orientation.lean

```diff
@@ -526,6 +580,38 @@ begin
     simp [h₁, h₂]
 end
 
+/-- If the index type has cardinality equal to the finite dimension, an orientation equals the
+negation of another orientation if and only if they are not equal. -/
+lemma ne_iff_eq_neg (x₁ x₂ : orientation R M ι) (h : fintype.card ι = finrank R M) :
+  x₁ ≠ x₂ ↔ x₁ = -x₂ :=
+⟨λ hn, (eq_or_eq_neg x₁ x₂ h).resolve_left hn, λ he, he.symm ▸ (module.ray.ne_neg_self x₂).symm⟩
+
+/-- The value of `orientation.map` when the index type has cardinality equal to the finite
+dimension, in terms of `f.det`. -/
+lemma map_eq_det_inv_smul (x : orientation R M ι) (f : M ≃ₗ[R] M)
```

Comment:
This lemma doesn't need a linear order, does it?

Reply:
We don't have a type class for non-linearly ordered fields, however. If enough of `linear_algebra.finite_dimensional` gets generalized to `module.free`, that should enable more general type class requirements here.